lib/agent.js

```diff
@@ -97,7 +109,7 @@ function dispatchFromAgent (requestType) {
       throw new InvalidReturnValueError(`Client returned from Agent.get() does not implement method ${requestType}`)
     }
 
-    return client[requestType]({ ...opts, method, path }, ...additionalArgs)
+    return client[requestType]({ ...opts, agent, requestType, method, path }, ...additionalArgs)
```

Comment:
I’m not sure I think this should be needed

Reply:
I need at least agent to being able to call dispatch in the right way. See my comment in the PR discussion.

Reply:
I'm now just adding agent (which might be originally given by the user and thus keeps the promise not to mutate opts)